usbsdmux/usbsdmux.py

```diff
@@ -18,16 +19,120 @@
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
+import os
 import time
 
-from .pca9536 import Pca9536
+from .i2c_gpio import Pca9536, Tca6408
 
 
-class UsbSdMux(object):
+class UnknownUsbSdMuxRevisionException(Exception):
+    pass
+
+
+def autoselect_driver(sg):
+    """
+    Create a new UsbSdMux with the correct driver for the device at /dev/<sg>
+
+    Arguments:
+    sg -- /dev/sg* to use
+    """
+
+    base_sg = os.path.realpath(sg)
+    sg_name = os.path.basename(base_sg)
+    model_filename = f"/sys/class/scsi_generic/{sg_name}/device/model"
+    try:
+        with open(model_filename, "r") as fh:
+            model = fh.read().strip()
+        if model == "sdmux HS-SD/MMC":
+            return UsbSdMuxLegacy(sg)
+        elif model == "sdFST HS-SD/MMC":
+            return UsbSdMuxFast(sg)
```

Comment:
`Legacy` has such a bad connotation. The current-gen USB-SD-Muxes are still fine, no need to get rid of them because they are `Legacy`.
And I am also not a huge fan of the new `FAST` name, but I think it is what it is now.

You can mark this as resolved. I just need this comment to link to it whenever I complain about the name :D.

Reply:
```suggestion
        if model == "sdmux HS-SD/MMC":
            return UsbSdMuxClassic(sg)
        elif model == "sdFST HS-SD/MMC":
            return UsbSdMuxFast(sg)
```

Reply:
Renamed `UsbSdMuxLegacy` to `UsbSdMuxClassic`.